src/index.js

```diff
@@ -514,31 +514,34 @@ var restart = function(sub, oldSub, dispatch) {
 }
 
 var start = function(sub, dispatch) {
-  return merge(sub, {
-    cancel: sub.effect(sub, dispatch)
-  })
+  return [
+    sub[0],
+    merge(sub[1], {
```

Comment:
So in V2 we provide all new state to the engine. Is there a reason we still have `merge`? No need to `for` through two objects and replace if we can just swap new state provided by the action right?

Just curious.

Reply:
> @selfup I would be in favor of something like `replace` instead of merge.

We totally are. V2's state change strategy is by replace, not by merge like in V1. This instance of `merge` is for a different thing. It's a way to add the `cancel` property to the subscription, so we know how to cancel it later.

Reply:
Yea I see that now. We use merge for checks mostly. I updated my comment 😄

Reply:
Yea I guess updating the object directly is impure. Nvm, merge makes sense, I was just hoping we wouldn't have to do all that work just to add cancel 😂

Reply:
Since we're already putting the sub function and props into an array, we could just add the cancel as another element if that's not too confusing to keep track of 🤷‍♂️

Reply:
I've got mixed feelings on that. We wanted a tuple data structure, which a third item violates, **but** it's not really a argument we should pass in and a subscription shouldn't be able to cancel itself.

Reply:
Tuples can have three members in a few languages, including Elm.

I'd prefer that to having to merge a cancel prop to my subscription props.

Reply:
I can certainly make the change of we're all behind it

Reply:
@mrozbarry Sounds good. If I understand this correctly, this is only an implementation detail, not a change to the API.

Reply:
Done